### PR TITLE
fix: remove ctx.cloud.reset in tests, handle infinite loop in stale results

### PR DESCRIPTION
**Pull request by tgriesser**
Related to #21250

### Additional details
Correctly fixes tests failing due to the introduction of the new queries on the specs page, properly handles an infinite loop that was occurring when an incomplete response was sent back for a `stale` fetch result. This would cause an infinite re-fetch loop while the server was looking to fulfill the stale value. Instead, now if we detect this situation we clear the urql cache in both the electron process and the client.

### Steps to test

Existing tests should be passing with only minor, expected modifications

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/graphql/src/stitching/remoteSchemaWrapped.ts [hunk 1]
```diff
@@ -65,13 +68,16 @@ export const remoteSchemaWrapped = wrapSchema<DataContext>({
...
-        obj.context?.graphql.pushResult({
+        context.graphql.pushResult({
           result: result?.[info.fieldName] ?? null,
           source: obj.rootValue,
           info,
-          ctx: obj.context,
+          ctx: context,
         })
       },
+      invalidateCache () {
```
Comment (tgriesser) [inline packages/graphql/src/stitching/remoteSchemaWrapped.ts:78]: Not super happy with the naming of this, might change to `onInvalidateCache`, and centralize the client & server invalidation calls as like `ctx.invalidateGraphQLCache()`.

File: packages/graphql/src/stitching/remoteSchemaWrapped.ts [hunk 2]
```diff
@@ -65,13 +68,16 @@ export const remoteSchemaWrapped = wrapSchema<DataContext>({
       // When we respond eagerly with a result, but receive an updated value
       // for the query, we can "push" the data down using the pushFragment subscription
       onUpdatedResult (result) {
-        obj.context?.graphql.pushResult({
```
Comment (mike-plummer) [inline packages/graphql/src/stitching/remoteSchemaWrapped.ts:68]: Looks like the TS typings are good with it, but the original code treated `context` as nullable. Any risk of that here?

Comment (tgriesser) [reply packages/graphql/src/stitching/remoteSchemaWrapped.ts:68]: No, because it's assigned above so it can't change

**Review by estrada9166 — APPROVED** [review]

File: packages/data-context/src/sources/CloudDataSource.ts [hunk 3]
```diff
@@ -48,9 +48,16 @@ export interface CloudExecuteQuery {
...
+  fieldName: string
   operationType?: OperationTypeNode
   requestPolicy?: RequestPolicy
   onUpdatedResult?: (data: any) => any
+  /**
+   * Triggered when we have an initial stale response that is not fulfilled
+   * by an additional fetch to the server. This means we've gotten into a bad state
+   * and we need to clear both the server & client side cache
+   */
+  invalidateCache: () => any
```
[the remark this answers is not in the document]
Comment (tgriesser) [reply packages/data-context/src/sources/CloudDataSource.ts:60]: Nah it should be void, will updated

**Pull request merged**